frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js

```diff
@@ -885,6 +896,26 @@ export default function lineAreaBar(
 
   checkSeriesIsValid(props);
 
+  if (props.chartType === "waterfall") {
+    props.series.map(series => {
```

Comment:
This should probably be `forEach` if we're just relying on it for iteration rather than the return values. Though, is there a good way to avoid mutation here?

Reply:
I've changed it so that the original series data need not be mutated anymore.